### PR TITLE
chore: bump amplify-android version to 1.16.13

### PR DESCRIPTION
**Pull request by haverchuck**
*Description of changes:*
Bumps amplify-android version to 1.16.13.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: packages/amplify_analytics_pinpoint/android/build.gradle [hunk 1]
```diff
@@ -60,8 +60,8 @@ dependencies {
     api amplifyCore
 
     implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
-    implementation 'com.amplifyframework:aws-analytics-pinpoint:1.6.10'
-    implementation 'com.amplifyframework:aws-auth-cognito:1.6.10'
+    implementation 'com.amplifyframework:aws-analytics-pinpoint:1.16.13'
+    implementation 'com.amplifyframework:aws-auth-cognito:1.16.13'
```
Comment (Ashish-Nanda) [inline packages/amplify_analytics_pinpoint/android/build.gradle:64]: Should probably be removed in the future.

File: packages/amplify_datastore/android/build.gradle [hunk 2]
```diff
@@ -57,8 +57,8 @@ android {
 
 dependencies {
     implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
-    implementation "com.amplifyframework:aws-datastore:1.6.10"
-    implementation "com.amplifyframework:aws-api-appsync:1.6.10"
+    implementation "com.amplifyframework:aws-datastore:1.16.13"
+    implementation "com.amplifyframework:aws-api-appsync:1.16.13"
```
Comment (Ashish-Nanda) [inline packages/amplify_datastore/android/build.gradle:61]: Should probably be removed in the future now that we have the Amplify Flutter API plugin.

File: packages/amplify_flutter/android/build.gradle [hunk 3]
```diff
@@ -58,13 +58,13 @@ android {
...
+    implementation 'com.amplifyframework:core:1.16.13'
     implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
     testImplementation 'junit:junit:4.13'
     testImplementation 'org.mockito:mockito-core:3.1.0'
     testImplementation 'org.mockito:mockito-inline:3.1.0'
     testImplementation 'androidx.test:core:1.3.0'
     testImplementation 'org.robolectric:robolectric:4.3.1'
     testImplementation 'com.google.code.gson:gson:2.8.6'
-    testImplementation 'com.amplifyframework:aws-auth-cognito:1.6.10'
+    testImplementation 'com.amplifyframework:aws-auth-cognito:1.16.13'
```
Comment (Ashish-Nanda) [inline packages/amplify_flutter/android/build.gradle:69]: Is this needed?

**Review by Ashish-Nanda — APPROVED** [review]
LGTM.

A number of `.gradle` files here have a dependency on the `auth-cognito` plugin, which should not be required. Customers should add that plugin through Flutter.

Can look into this in a followup PR.

File: packages/amplify_flutter/android/build.gradle (hunk 3 above)
Comment (haverchuck) [reply packages/amplify_flutter/android/build.gradle:69]: I don't think this is needed - it's a test dependency, so we can remove it later without problems.

**Review by Amplifiyer — APPROVED** [review]

**Pull request merged**